src/Features/Core/Portable/Completion/Providers/AbstractRecommendationServiceBasedCompletionProvider.cs

```diff
@@ -40,30 +43,30 @@ internal abstract class AbstractRecommendationServiceBasedCompletionProvider<TSy
                 var taskType = context.SemanticModel.Compilation.TaskType();
                 return recommendedSymbols.NamedSymbols.SelectAsArray(
                     s => IsValidForTaskLikeTypeOnlyContext(s, context),
-                    s => (s, preselect: s.OriginalDefinition.Equals(taskType)));
+                    s => new SymbolAndSelectionInfo(Symbol: s, Preselect: s.OriginalDefinition.Equals(taskType)));
             }
             else if (context.IsGenericConstraintContext)
             {
                 // Just filter valid symbols. Nothing to preselect
-                return recommendedSymbols.NamedSymbols.SelectAsArray(IsValidForGenericConstraintContext, s => (s, preselect: false));
+                return recommendedSymbols.NamedSymbols.SelectAsArray(IsValidForGenericConstraintContext, s => new SymbolAndSelectionInfo(Symbol: s, Preselect: false));
             }
             else
             {
                 var shouldPreselectInferredTypes = await ShouldPreselectInferredTypesAsync(completionContext, position, options, cancellationToken).ConfigureAwait(false);
                 if (!shouldPreselectInferredTypes)
-                    return recommendedSymbols.NamedSymbols.SelectAsArray(s => (s, preselect: false));
+                    return recommendedSymbols.NamedSymbols.SelectAsArray(s => new SymbolAndSelectionInfo(Symbol: s, Preselect: false));
 
                 var inferredTypes = context.InferredTypes.Where(t => t.SpecialType != SpecialType.System_Void).ToSet();
 
-                using var _ = ArrayBuilder<(ISymbol symbol, bool preselect)>.GetInstance(out var result);
+                using var _ = ArrayBuilder<SymbolAndSelectionInfo>.GetInstance(out var result);
 
                 foreach (var symbol in recommendedSymbols.NamedSymbols)
                 {
                     // Don't preselect intrinsic type symbols so we can preselect their keywords instead. We will also
                     // ignore nullability for purposes of preselection -- if a method is returning a string? but we've
                     // inferred we're assigning to a string or vice versa we'll still count those as the same.
                     var preselect = inferredTypes.Contains(GetSymbolType(symbol), SymbolEqualityComparer.Default) && !IsInstrinsic(symbol);
-                    result.Add((symbol, preselect));
+                    result.Add(new(symbol, preselect));
```

Comment:
inconsistent use of implicit-object.  given howlightweight this type is, i'm ok with you using the implicit syntax everywhere for this.